### PR TITLE
Mailbox - log on over capacity

### PR DESCRIPTION
**Pull request by PiotrTrzpil**

### CODE REVIEW

File: core/services/head_broadcaster.go [hunk 1]
```diff
@@ -83,15 +83,18 @@ func (hr *HeadBroadcaster) Connect(head *models.Head) error {
 func (hr *HeadBroadcaster) Disconnect() {}
 
 func (hr *HeadBroadcaster) OnNewLongestChain(ctx context.Context, head models.Head) {
-	hr.mailbox.Deliver(head)
+	wasOverCapacity := hr.mailbox.Deliver(head)
+	if wasOverCapacity {
+		logger.Error("HeadBroadcaster: head mailbox is over capacity - dropped the oldest unprocessed head")
```
Comment (RyanRHall) [inline core/services/head_broadcaster.go:88]: This probably shouldn't be an `Error` level log. I would make it `Debug`, as this is common and expected and not a reason for alarm.

File: core/services/bulletprooftxmanager/eth_confirmer.go [hunk 2]
```diff
@@ -98,7 +98,10 @@ func (ec *ethConfirmer) Disconnect() {
 // - Now we finish head 41
 // - We move straight on to processing head 45
 func (ec *ethConfirmer) OnNewLongestChain(ctx context.Context, head models.Head) {
-	ec.mb.Deliver(head)
+	wasOverCapacity := ec.mb.Deliver(head)
+	if wasOverCapacity {
+		logger.Error("EthConfirmer: head mailbox is over capacity - dropped the oldest unprocessed head")
```
Comment (RyanRHall) [inline core/services/bulletprooftxmanager/eth_confirmer.go:103]: Maybe check with @samsondav how important this is. I would think `Warn`.

File: core/services/gasupdater/gas_updater.go [hunk 3]
```diff
@@ -110,7 +110,10 @@ func (gu *gasUpdater) Disconnect() {
 // OnNewLongestChain recalculates and sets global gas price if a new head comes
 // in and we are not currently fetching
 func (gu *gasUpdater) OnNewLongestChain(ctx context.Context, head models.Head) {
-	gu.mb.Deliver(head)
+	wasOverCapacity := gu.mb.Deliver(head)
+	if wasOverCapacity {
+		logger.Error("GasUpdater: head mailbox is over capacity - dropped the oldest unprocessed head")
```
Comment (RyanRHall) [inline core/services/gasupdater/gas_updater.go:115]: Same comment here. Error level logging might be overkill.

File: core/services/keeper/registry_synchronizer_log_listener.go [hunk 4]
```diff
@@ -39,18 +39,28 @@ func (rs *RegistrySynchronizer) HandleLog(broadcast log.Broadcast) {
...
+		mailboxName = "mbUpkeepRegistered"
 	case *keeper_registry_wrapper.KeeperRegistryUpkeepPerformed:
-		rs.mailRoom.mbUpkeepPerformed.Deliver(broadcast)
+		wasOverCapacity = rs.mailRoom.mbUpkeepPerformed.Deliver(broadcast)
+		mailboxName = "mbUpkeepPerformed"
 	default:
 		logger.Warnf("unexpected log type %T", log)
 	}
+	if wasOverCapacity {
+		logger.Errorf("RegistrySynchronizer: %v mailbox is over capacity - dropped the oldest unprocessed item", mailboxName)
```
Comment (RyanRHall) [inline core/services/keeper/registry_synchronizer_log_listener.go:64]: `ERROR` seems appropriate here :+1:

File: core/services/keeper/upkeep_executer.go [hunk 5]
```diff
@@ -99,7 +99,10 @@ func (executer *UpkeepExecuter) Close() error {
 }
 
 func (executer *UpkeepExecuter) OnNewLongestChain(ctx context.Context, head models.Head) {
-	executer.mailbox.Deliver(head)
+	wasOverCapacity := executer.mailbox.Deliver(head)
+	if wasOverCapacity {
```
Comment (RyanRHall) [inline core/services/keeper/upkeep_executer.go:103]: not important. We can delete entirely here.

**Review by RyanRHall — CHANGES_REQUESTED** [review]

**Comment by PiotrTrzpil** [comment]
@RyanRHall ok makes sense, thanks - i suppose even on a temporary node overload, Error might be a bit too strong

File: core/services/bulletprooftxmanager/eth_confirmer.go (hunk 2 above)
Comment (PiotrTrzpil) [reply core/services/bulletprooftxmanager/eth_confirmer.go:103]: True, i checked that it's only used a limit, so each head is not needed directly

File: core/services/head_broadcaster.go (hunk 1 above)
Comment (PiotrTrzpil) [reply core/services/head_broadcaster.go:88]: oh right, it has capacity of 1. So i'll remove this logging

**Review by samsondav — COMMENTED** [review]
I'm ok with error level for now, none of these capacities should ever be exceeded in normal operation and can have harmful consequences if it does happen if I understand correctly?

**Review by samsondav — APPROVED** [review]

**Pull request merged**